### PR TITLE
[BUGFIX] Add `<head>` element despite `<header>`

### PR DESCRIPTION
**Pull request by JakeQZ**
Also test around the issue.

Resolves #844.

### CODE REVIEW

File: tests/Unit/CssInlinerTest.php [hunk 1]
```diff
@@ -2252,6 +2252,20 @@ public function inlineCssAppliesCssToMatchingElementsAndKeepsRuleWithPseudoCompo
         self::assertContainsCss('<style type="text/css">a:hover { color: green; }</style>', $result);
     }
 
+    /**
+     * #844
```
Comment (oliverklee) [inline tests/Unit/CssInlinerTest.php]: To keep the code succinct, please let's not have issue numbers in the code. Having them in the changelog and the git history should suffice.

File: tests/Unit/HtmlProcessor/AbstractHtmlProcessorTest.php [hunk 2]
```diff
@@ -175,6 +175,9 @@ public function contentWithoutHeadTagDataProvider(): array
             'doctype only' => ['<!DOCTYPE html>'],
             'body content only' => ['<p>Hello</p>'],
             'BODY element' => ['<body></body>'],
+            // #844
```
Comment (oliverklee) [inline tests/Unit/HtmlProcessor/AbstractHtmlProcessorTest.php]: Please let's remove the issue number here as well.

**Comment by JakeQZ** [comment]
> To keep the code succinct, please let's not have issue numbers in the code. Having them in the changelog and the git history should suffice.

OK, I've removed these.

Perhaps I am abusing the `needs unit tests` label?  I added it because the tests added went slightly beyond the specific issue.  But perhaps it is intended as a temporary label like `waiting for feedback`?

**Comment by oliverklee** [comment]
@JakeQZ I intended the `unit tests needed`label to mean "there still are some tests missing".

**Pull request merged**